### ISSUE TITLE
DATABASE_URL verification still attempted when datasource override provided in PrismaClient constructor

### ISSUE THREAD

**Issue by timleslie** [prisma/prisma#3750]
## Bug description

https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-client/constructor#datasources

When attempting to connect to the prisma client which is configured with an explicit datasource location:

```javascript
const prisma = new PrismaClient({
  datasources: { postgresql: { url: "postgres://****:****@localhost:5432/****?schema=foo" } },
});
...
await prisma.$connect();
```

I get the following error:

```
(node:99546) UnhandledPromiseRejectionWarning: Error: error: Environment variable not found: DATABASE_URL.
  -->  schema.prisma:5
   | 
 4 | datasource postgresql {
 5 |   url      = env("DATABASE_URL")
   | 

Validation Error Count: 1
    at Oe.Yu.default.on.c (/Users/timleslie/src/prisma-issue/node_modules/@prisma/client/engine-core/src/NodeEngine.ts:591:29)
    at Oe.emit (events.js:197:13)
    at Oe.EventEmitter.emit (domain.js:439:20)
    at addChunk (_stream_readable.js:288:12)
    at readableAddChunk (_stream_readable.js:269:11)
    at Oe.Readable.push (_stream_readable.js:224:10)
    at Oe.Transform.push (_stream_transform.js:151:32)
    at Oe._pushBuffer (/Users/timleslie/src/prisma-issue/node_modules/@prisma/client/engine-core/src/byline.ts:116:17)
    at Oe._transform (/Users/timleslie/src/prisma-issue/node_modules/@prisma/client/engine-core/src/byline.ts:107:8)
    at Oe.Transform._read (_stream_transform.js:190:10)
(node:99546) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:99546) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

## How to reproduce

```
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

datasource postgresql {
  url      = env("DATABASE_URL")
  provider = "postgresql"
}

generator client {
  provider = "prisma-client-js"
}

model User {
  id       Int     @id @default(autoincrement())
  name     String?
}
```

## Expected behavior

I would expect the client to never look for the `DATABASE_URL` environment variable if the override has been provided.

## Environment & setup

- OS: OSX 10.15.6
- Database: PostgreSQL 12.2
- Node.js version: 12.11.0
- Prisma version: 2.7.1

**Comment by janpio** [comment]
Does it work if you set the `DATABASE_URL`? Then the title could be adapted that it unexpectedly still tries to verify the ENV var in the schema file instead of only using the one that is supplied in the constructor directly.

**Comment by timleslie** [comment]
Yes, it does work when I set `DATABASE_URL`. I'll update the issue title accordingly 👍

**Comment by timleslie** [comment]
Hi @pantharshit00, is there any movement on this issue?

**Comment by divyenduz** [comment]
@timleslie Thanks for raising this again. I am able to reproduce this with `2.10.0-dev.48`. This is not a regression, the same behaviour can be observed in `2.0.0`. 

Raising this internally for a quick fix :)

**Comment by timsuchanek** [comment]
In this case, we are providing the `OVERWRITE_DATASOURCES` env var to the query engine, which should override the datasource and therefore skip the env check for `DATABASE_URL`

**Comment by pantharshit00** [comment]
This is still reproducible in `2.11.0-dev.9`

![image](https://user-images.githubusercontent.com/22195362/97689629-dfc25f80-1ac1-11eb-93eb-e176443a07cf.png)

**Comment by DustinJSilk** [comment]
Any new updates on this??

**Comment by pantharshit00** [comment]
@DustinJSilk We are aware of this problem and we intent to tackle it in this sprint(bi weekly release schedule). 

There are lot of pending issues so I wouldn't guarantee it will 100% make it in the next release but we are aware that this is pain point.

**Comment by DustinJSilk** [comment]
Thanks for the speedy response @pantharshit00. I've figured out a temporary fix on my side so no problem if it doesn't make the nest release.

**Comment by dpetrick** [comment]
This is fixed an will be in the 2.11 release.
